app.js

```diff
@@ -34,6 +35,7 @@ function clientLoaded (client) {
   client.on('StasisStart', function(event, channel) {
     if (!isDialed(event.args[0])) {
       var extension = event.args[0];
+      var confFile = process.argv[2];
       sla(client, confFile, channel, extension)
```

Comment:
To my knowledge, channel.on('ChannelHold') is all that you need in order to get those events. I am not entirely sure what you mean by applying the HOLD_INTERCEPT function either.